components/tools/OmeroJava/test/integration/gateway/TablesFacilityTest.java

```diff
@@ -136,75 +84,90 @@ public void testBigTable() throws Exception {
-                    if (type.equals(String.class)) {
-                        Assert.assertEquals((String)data2[c][r-(int)td2.getOffset()], (String)data[index][r]);
-                    } else if (type.equals(Long.class)) {
-                        Assert.assertEquals((Long)data2[c][r-(int)td2.getOffset()], (Long)data[index][r]);
-                    } else if (type.equals(Double.class)) {
-                        Assert.assertEquals((Double)data2[c][r-(int)td2.getOffset()], (Double)data[index][r]);
-                    } else if (type.equals(Double[].class)) {
-                        Assert.assertEquals((Double[])data2[c][r-(int)td2.getOffset()], (Double[])data[index][r]);
-                    }
+
+        original.setOriginalFileId(stored.getOriginalFileId());
+
+        System.out.println("Storing table with " + (nCols * nRows)
+                + " entries took " + duration + " ms");
+    }
+
+    @Test(dependsOnMethods = { "testAddTable" })
+    public void testGetTableInfo() throws Exception {
+        TableData info = tablesFacility.getTableInfo(rootCtx,
+                original.getOriginalFileId());
+        Assert.assertEquals(info.getNumberOfRows(), nRows);
+        Assert.assertEquals(info.getColumns(), original.getColumns());
+    }
+
+    @Test(dependsOnMethods = { "testAddTable" })
+    public void testGetAvailableTables() throws Exception {
+        Collection<FileAnnotationData> tablesFiles = tablesFacility
+                .getAvailableTables(rootCtx, ds);
+        Assert.assertEquals(1, tablesFiles.size());
+        Assert.assertEquals(tablesFiles.iterator().next().getFileID(),
+                original.getOriginalFileId());
+    }
+
+    @Test(dependsOnMethods = { "testAddTable" }, invocationCount = 10, threadPoolSize = 5)
+    /**
+     * Read a random subset from the table and compare to the original data
+     * 
+     * @throws Exception
+     */
+    public void testReadBigTable() throws Exception {
+        Object[][] origData = original.getData();
+
+        TableData info = tablesFacility.getTableInfo(rootCtx,
+                original.getOriginalFileId());
+        int rows = (int) info.getNumberOfRows();
+        int cols = info.getColumns().length;
+
+        int rowFrom = rand.nextInt(rows - 1000); // can't request more than 1000
+                                                 // rows at once
+        int rowTo = rowFrom + rand.nextInt(1000);
+        int[] columns = new int[rand.nextInt(cols)];
+
+        for (int x = 0; x < columns.length; x++)
+            columns[x] = rand.nextInt(cols);
+
+        long start = System.currentTimeMillis();
+        TableData td2 = tablesFacility.getTable(rootCtx,
+                original.getOriginalFileId(), rowFrom, rowTo, columns);
+        System.out.println("Requesting random subset of size: "
+                + ((rowTo - rowFrom) * columns.length) + " entries, took "
+                + (System.currentTimeMillis() - start) + " ms");
```

Comment:
maybe could mention more as a "from, to" explicitly giving `rowFrom`, `rowTo`: if the test fails, there might then be enough information on the console to reproduce the problem